apps/meteor/client/views/room/providers/SelectedMessagesProvider.tsx

```diff
@@ -61,13 +84,24 @@ export const selectedMessageStore = new (class SelectMessageStore extends Emitte
 		this.isSelecting = false;
 		this.emit('toggleIsSelecting', false);
 	}
+
+	toggleAll(mids: string[]): void {
+		mids.forEach((mid) => this.select(mid));
+	}
 })();
 
 type SelectedMessagesProviderProps = {
 	children?: ReactNode;
 };
 
 export const SelectedMessagesProvider = ({ children }: SelectedMessagesProviderProps) => {
+	const room = useRoom();
+	const messages = useMessages({ rid: room._id });
+
+	useEffect(() => {
+		selectedMessageStore.availableMessages = new Set(messages.map((message) => message._id));
```

Comment:
Since the store is a class, instead of assigning a value directly to to a property, a method that receives an array of messages would be best for this use.

But I think we can improve a little on this solution so that we don't even need this useEffect. Each rendered message component calls `useIsSelectedMessage` with the messageId. You could register the available messages from inside the hook, then you don't need to keep syncing this value in the Provider every time something changes...